### PR TITLE
chore: remove package-lock.json and switch from npm ci to npm install

### PR DESCRIPTION
**Pull request by ankita10119**
### Description

Removes `package-lock.json` from the repository to avoid committing internal registry URLs (e.g. JFrog) to the public repo.                                                                                                                        
   
The lockfile is not shipped in the npm package (`files: ["dist"]` in `package.json` ensures only compiled output is published), so this has no impact on consumers of this library.                                                                                                                                

#### CI pipeline changes

**`cache: npm` → manual `~/.npm` cache**
`cache: npm` on `actions/setup-node` requires a lockfile. Replaced with `actions/cache@v4` targeting `~/.npm`, keyed on a hash of all `package.json` files. This preserves install performance without needing a lockfile.

**`npm ci` → `npm install`**
`npm ci` requires a lockfile. Switched to `npm install` across build and publish actions.

**Accepted risk — publish pipeline**
Without a lockfile, the publish pipeline resolves latest compatible dependency versions at release time. For a library where all dependencies are `devDependencies` (not shipped to consumers), this risk is accepted. Runtime behaviour for library users is unaffected.

#### Rollup toolchain upgrade

Removing the lockfile caused npm to resolve `typescript@5.9.3` (latest), which broke `rollup-plugin-dts@4` whose peer dependency is `typescript@^4.1`.

The following upgrades were required to restore build compatibility:

  - `rollup` `^2` → `^3`
  - `rollup-plugin-dts` `^4` → `^5` (adds TypeScript 5.x peer dep support)
  - `@rollup/plugin-node-resolve` `^13` → `^15`
  - Replaced `rollup-plugin-typescript2` with `@rollup/plugin-typescript ^11.1.6`
  - Updated `rollup.config.js` to use the new plugin API (`compilerOptions` directly instead of `tsconfigOverride`)
  - Removed `type` keyword from 4 imports in `src/index.ts` — required by `@rollup/plugin-typescript`
  - Removed `rollup-plugin-typescript2` from `package.json` — no longer used

These changes are inseparable from the lockfile removal: they only surface because the lockfile was previously pinning compatible versions.

### References

Supersedes #228 

### Testing
- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not the default branch

### CODE REVIEW

File: scripts/strip-lock-resolved.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  throw e;
+}
+
+try {
+  const lock = JSON.parse(fs.readFileSync(fd, 'utf8'));
+
+  // lockfile v2/v3 — flat packages map
+  // Preserve resolved on link:true entries — these are workspace symlinks pointing
+  // to local package paths, not registry URLs.
+  if (lock.packages) {
```
Comment (nandan-bhat) [inline scripts/strip-lock-resolved.js]: When processing lockfile v1 (`lock.dependencies`), the script may delete `resolved` from every dependency without checking for `link: true`. In a workspace monorepo, some dependencies may be workspace symlinks that rely on the `resolved` field; deleting it could cause issues. The v2/v3 handling reportedly includes this guard but v1 does not.

> Add a guard similar to the v2/v3 handling: only delete `resolved` if the dependency is not a workspace link (e.g., `if (!dep.link) delete dep.resolved;`).

File: scripts/strip-lock-resolved.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
```
Comment (nandan-bhat) [inline scripts/strip-lock-resolved.js]: No automated tests are added for the new `strip-lock-resolved.js` script. Without tests, regressions in lockfile handling (e.g., breaking workspace links, handling edge cases in different lockfile versions) may go unnoticed.

> Add unit tests that load sample lockfiles for v1, v2, and v3 formats, verify that `resolved` fields are removed for registry packages but preserved for `link: true` entries, and ensure the script handles missing `package-lock.json` gracefully.

**Comment by nandan-bhat** [comment]
@ankita10119 Why aren't we just removing the `package-lock.json` file ? Why do we need it ?

File: scripts/strip-lock-resolved.js (hunk 1 above)
Comment (ankita10119) [reply scripts/strip-lock-resolved.js]: Addressed and updated

File: scripts/strip-lock-resolved.js (hunk 2 above)
Comment (ankita10119) [reply scripts/strip-lock-resolved.js]: Addressed and updated

**Comment by ankita10119** [comment]
> @ankita10119 Why aren't we just removing the `package-lock.json` file ? Why do we need it ?

I actually reconsidered and this PR has been updated to do exactly that: **remove package-lock.json entirely**                                        
                                                                                                                                                                        
Initially, I took the approach of keeping `package-lock.json` but stripping the resolved fields (registry URLs) via a pre-commit hook. 
The reasoning was to preserve the benefits of a lockfile, deterministic installs, npm ci, Dependabot transitive dependency tracking, and a stable publish pipeline.  
                                                                                                                                                                        
However, upon analysing further, I realised that unlike application repos, this SDK does not ship `node_modules` or `package-lock.json` to consumers. The files: ["dist"] field in package.json ensures only the compiled output is published to npm.
This means `package-lock.json` exists purely for the GitHub repo and the CI/CD pipeline, it has no role in bundling or what end users receive. Given that, it is safe to remove it entirely and update the pipeline accordingly, without any impact on consumers of this library.

The strip-resolved approach was also overly complex and introduced a monorepo-specific edge case: workspace symlink entries in the lockfile use resolved to store local package paths (not registry URLs), which had to be preserved separately. Removing the lockfile entirely avoids all of that complexity.

This PR now follows the same approach as #228, but with all you PR review comments from that PR addressed:
  - `cache: npm` replaced with manual `~/.npm` caching (addresses the caching concern)
  - `npm install` used in place of `npm ci` throughout
  - Rollup toolchain upgraded to fix the TypeScript 5.x peer dependency conflict that surfaces without a lockfile (addresses the mixed-concerns question, these changes are forced by the lockfile removal)
  - `rollup-plugin-typescript2` removed as it is no longer used
  - type keyword removal in `src/index.ts` explained as a requirement of `@rollup/plugin-typescript`
  - Publish pipeline non-determinism called out explicitly as an accepted risk in the PR description

  The full rationale is documented in the updated PR description above

**Comment by nandan-bhat** [comment]
Add `package-lock.json` to `.gitignore`

**Review by nandan-bhat — DISMISSED** [review]

**Review by nandan-bhat — APPROVED** [review]
